specification/communication/data-plane/PhoneNumbers/preview/2023-10-01-preview/phonenumbers.json

```diff
@@ -530,6 +583,325 @@
+            "format": "int32",
+            "default": 100
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A collection of Phone Number Reservations, and an optional link to the next page of results.",
+            "schema": {
+              "$ref": "#/definitions/PhoneNumbersReservations"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2022-07-13/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "reservations"
+        },
+        "x-ms-examples": {
+          "Get reservations": {
+            "$ref": "./examples/GetReservations.json"
+          }
+        }
+      }
+    },
+    "/availablePhoneNumbers/reservations/{reservationId}": {
+      "put": {
+        "tags": [
+          "Browse, reserve and purchase available phone numbers"
+        ],
+        "summary": "Initializes a new reservation with a given ID. By default the new reservation is empty, but it can optionally be created with an initial list of phone numbers.",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "operationId": "PhoneNumbers_CreateReservation",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "reservationId",
+            "description": "The id of the reservation.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "body",
+            "name": "reservation",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/PhoneNumbersReservation"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Reservation successfully created.",
+            "schema": {
+              "$ref": "#/definitions/PhoneNumbersReservation"
+            }
+          },
+          "409": {
+            "description": "Conflict while creating reservation. This can happen if a reservation with the same ID already exists and doesn't have 'active' status.",
+            "schema": {
+              "$ref": "../../../Common/stable/2022-07-13/common.json#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2022-07-13/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create reservation": {
+            "$ref": "./examples/CreateReservation.json"
+          }
+        }
+      },
+      "patch": {
```

Comment:
Why `PUT` and `PATCH`? Generally, we recommend `PATCH` for create-or-update semantics. See https://aka.ms/azapi/guidelines